File: src/main/java/kr/co/wanted/market/member/repository/MemberRepository.java
package kr.co.wanted.market.member.repository;

import kr.co.wanted.market.member.entity.Member;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {

    Optional<Member> findByLoginId(String id);

}

File: src/main/kotlin/wanted/market/member/repository/MemberRepository.kt
package wanted.market.member.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import wanted.market.member.entity.Member
import java.util.*

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByEmail(email: String): Optional<Member>
}